### PR TITLE
feat: set output parameter wayback_url upon save attempt

### PR DESCRIPTION
**Pull request by jonblatho**
Reattempting because I shouldn't have recompiled on the last PR attempt. Apologies!

----

I found it handy for a project in which I'm using this action to set an output parameter upon a save attempt to either an empty string or the Wayback Machine URL depending on whether the save was successful, so that the URL is more easily available to other steps.

I'm new to JavaScript/GitHub Actions development so I'm sure this needs edits/refactoring, but I figured I'd make my work available for implementation as it might be useful for someone else.

### CODE REVIEW

File: src/wayback.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import * as core from '@actions/core';
```
Comment (JamieMagee) [inline src/wayback.ts:1]: I've got `allowSyntheticDefaultImports` set in `tsconfig.json`, so you should be able to do

```suggestion
import core from '@actions/core';
```

**Comment by JamieMagee** [comment]
Thanks for the contribution. I've left a couple of comments and suggestions in the code, and I've got a couple more question:

1. What version of yarn did you use? The changes in `yarn.lock` look like churn and seem to be unnecessary
2. Can you add a simple test for this in `test/wayback.spec.ts`?

**Comment by jonblatho** [comment]
No problem.

1. All I can see is `"yarn": ">=1.22.0"` in `package-lock.json`, which appears to be the final minor release for yarn 1.x. I'm not familiar with yarn but cursory research suggested that either `yarn.lock` or `package-lock.json` should be committed for projects that have both, so I committed it since it was already included in the repository. I can certainly roll it back.
2. I'll give it a shot. It'll definitely need a second set of eyes though.

**Comment by jonblatho** [comment]
1. It appears that rolling back the yarn.lock file to the latest one here broke it. I still get this error even when using the exact version that appears to be running in the CI builds, v1.22.5. It does work on my machine if I run `yarn install --frozen-lockfile --force`.
2. After looking at the docs for `@actions/core`, the only way of testing this is to monitor stdout calls to ensure that one is made that we'd expect. A model for this is [here](https://github.com/actions/toolkit/blob/main/packages/core/__tests__/core.test.ts#L303), but I'm unable to run tests locally to ensure they pass before committing because I get the following errors when attempting to run the exact `yarn jest --maxWorkers=2 --ci --coverage` call that is run in the CI builds:
```
  ●  › works

    TypeError: Cannot read property 'headers' of undefined

      61 |   private handleErrorResponse(response: AxiosResponse): void {
      62 |     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
    > 63 |     const error: string = response.headers?.['x-archive-wayback-runtime-error'];
         |                                    ^
      64 |     if (error) {
      65 |       switch (error) {
      66 |         case 'AdministrativeAccessControlException':

      at WayBack.handleErrorResponse (src/wayback.ts:63:36)
      at WayBack.save (src/wayback.ts:55:12)
      at Object.<anonymous> (test/wayback.spec.ts:34:5)

  ●  › polls

    TypeError: Cannot read property 'headers' of undefined

      61 |   private handleErrorResponse(response: AxiosResponse): void {
      62 |     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
    > 63 |     const error: string = response.headers?.['x-archive-wayback-runtime-error'];
         |                                    ^
      64 |     if (error) {
      65 |       switch (error) {
      66 |         case 'AdministrativeAccessControlException':

      at WayBack.handleErrorResponse (src/wayback.ts:63:36)
      at WayBack.save (src/wayback.ts:55:12)
      at Object.<anonymous> (test/wayback.spec.ts:47:5)
```
The same issue happened when I attempted to write a test on the updated getArchiveUrl function. So, long story short, I have no clue how to test this because tests on existing unchanged code aren't passing.

**Comment by JamieMagee** [comment]
Great deep dive on the testing status. I think I've got another solution though. We're not interested in whether or not `process.stdout.write` gets called, just that `core.setOutput` is. So we can utilize the mock in `test/setup.ts` and add an additional mock for `core.setOutput`, then check the status of that.

I'm not too sure what's going on with yarn. I checked what I get locally with what's in your branch, and I wasn't able to get the same results. I regenerated the lockfile on my end and committed it.

**Review by JamieMagee — APPROVED** [review]

**Comment by jonblatho** [comment]
Thank you for fixing that up for me! Hopefully it wasn't too much work; if it was, my apologies. Some of this stuff is way out of my depth but I figured it might prove useful to someone else like it did to me! Thanks again.

**Comment by JamieMagee** [comment]
Not a problem. Thanks for the contribution, and I'm glad you're finding it useful!

**Pull request merged**